marbot-aws-account-connection.yml

```diff
@@ -139,6 +146,83 @@ Resources:
           Condition:
             StringEquals:
               'sts:ExternalId': !Select [2, !Split ['/', !Ref 'AWS::StackId']]
+  MonitoringAssistantPolicy:
+    Condition: MonitoringAssistantEnabled
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyDocument:
+        Statement:
+        - Effect: Allow
+          Action:
+          - 'autoscaling:DescribeAutoScalingGroups'
+          - 'cloudwatch:DeleteAlarms'
+          - 'cloudwatch:DescribeAlarms'
+          - 'cloudwatch:ListTagsForResource'
+          - 'cloudwatch:PutMetricAlarm'
+          - 'ec2:DescribeAccountAttributes'
+          - 'ec2:DescribeInstanceAttribute'
+          - 'ec2:DescribeInstanceCreditSpecifications'
+          - 'ec2:DescribeInstances'
+          - 'ec2:DescribeInstanceTypes'
+          - 'ec2:DescribeVolumes'
+          - 'ec2:DescribeVpcs'
+          - 'elasticloadbalancing:DescribeLoadBalancers'
+          - 'elasticloadbalancing:DescribeTargetGroups'
+          - 'events:DeleteRule'
+          - 'events:DescribeRule'
+          - 'events:ListTagsForResource'
+          - 'events:ListTargetsByRule'
+          - 'events:PutRule'
+          - 'events:PutTargets'
+          - 'events:RemoveTargets'
+          - 'iam:CreateRole'
```

Comment:
Should we use permission boundaries to mitigate the risk of granting full administrator access?

Reply:
But how do we deploy the permission boundary? If you trust us creating the permission boundary role you can also trust us creating the roles itself?

Reply:
In theory, the customer is able to review this template before creating the stack. So, it is possible to review the permission boundary. Same is true for updating the stack.

But when marbot uses the IAM role to deploy the monitoring configuration the customer has no chance to review the IAM changes.

Reply:
I see